### PR TITLE
Get rid of entry settings object usage

### LINKED ISSUE #86: Remove usage of "entry settings object"
It seems most can be replaced with "relevant settings object" easily, but not entirely sure about usage in https://notifications.spec.whatwg.org/#create-a-notification.

### PR DESCRIPTION
**Pull request by annevk**
Use the relevant settings object instead.

Fixes #86.

(Note that this builds upon #167.)

***
<a href="https://whatpr.org/notifications/168.html" title="Last updated on Apr 26, 2021, 6:46 AM UTC (c7f1660)">Preview</a> | <a href="https://whatpr.org/notifications/168/a3e44ea...c7f1660.html" title="Last updated on Apr 26, 2021, 6:46 AM UTC (c7f1660)">Diff</a>

### CODE REVIEW

File: notifications.bs [hunk 1]
```diff
@@ -150,114 +158,109 @@ clipped corners.
...
+ <var>options</var>["{{NotificationOptions/body}}"].
 
-  <li><p>Set <var>notification</var>'s <a for=notification>tag</a> to
-  <var>options</var>["{{NotificationOptions/tag}}"].
+ <li><p>Set <var>notification</var>'s <a for=notification>tag</a> to
+ <var>options</var>["{{NotificationOptions/tag}}"].
 
-  <li><p>Let <var>baseURL</var> be the API base URL specified by the
-  <a>entry settings object</a>. <span class=XXX>Or incumbent?</span>
+ <li><p>Let <var>baseURL</var> be the API base URL specified by <var>settings</var>.
```
Comment (domenic) [inline notifications.bs:206]: This could be `<var>settings</var>'s <a for="environment settings object">API base URL</a>`

File: notifications.bs [hunk 2]
```diff
@@ -718,7 +722,7 @@ constructor steps are:
 <h3 id=static-members>Static members</h3>
 
 <p>The static <dfn attribute for=Notification><code>permission</code></dfn> getter steps are to
-return the <a>permission</a> for the <a>entry settings object</a>'s
+return the <a>permission</a> for <a>this</a>'s <a>relevant settings object</a>'s
```
Comment (domenic) [inline notifications.bs:725]: This is static so it doesn't work. You want the `<a>current settings object</a>`.

File: notifications.bs [hunk 3]
```diff
@@ -696,7 +699,8 @@ constructor steps are:
  <a>throw</a> a {{TypeError}}.
 
  <li><p>Let <var>notification</var> be the result of <a>creating a notification</a> given
- <var>title</var> and <var>options</var>. Rethrow any exceptions.
+ <var>title</var>, <var>options</var>, and <a>this</a>'s <a>relevant settings object</a>.
```
Comment (domenic) [inline notifications.bs:702]: They're equivalent in constructors, but using this's relevant settings object here and current settings object a few lines up is a bit confusing.

File: notifications.bs (hunk 2 above)
Comment (domenic) [reply notifications.bs:725]: Hmm this probably is preexisting in requestPermission() somehow.

**Review by domenic — CHANGES_REQUESTED** [review]
Statics are problematic

File: notifications.bs (hunk 2 above)
Comment (annevk) [reply notifications.bs]: I fixed both, thanks for spotting that!

**Comment by annevk** [comment]
New commit message:
```
Get rid of entry settings object usage

Use the relevant global/settings object where possible; use the current global/settings object in statics.

Fixes #86.
```

**Review by domenic — APPROVED** [review]

**Pull request merged**